ClrMD.Extensions/ClrDynamic.cs

```diff
@@ -60,7 +61,8 @@ public class ClrDynamic : DynamicObject, IEnumerable<ClrDynamic>, IComparable, I
                 if (!Type.IsArray)
                     throw new InvalidOperationException(string.Format("Type '{0}' is not an array", Type.Name));
 
-                int arrayLength = Type.GetArrayLength(Address);
+                //int arrayLength = Type.GetArrayLength(Address);
+                int arrayLength = Heap.GetObject(Address).AsArray().Length;
```

Comment:
Is there a way to get the array length without allocating a new array with `AsArray()`? The current statement could lead to poor performance.